### PR TITLE
Created a prototype of the OLMo Class for regression and classification tasks

### PR DESCRIPTION
**Pull request by priyadp1**
## Description
Added a OLMo class with regression and classification tasks with 2 pytest files for both regression and classification. The generation task isn't necessary since it's already being handled in hf_models.py and including the generate() function in the OLMo class would be redundant. For regression and classification, a task-specific linear head is attached to the model, and the workflow is essentially: OLMo hidden state -> self.model.score -> logits.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [X] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [X] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [X ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Zrahay** [comment]
Hi Prisha! I am not a reviewer of your code but just noticed one small mistake. You can remove that Fix#(Issue) line from your PR for a cleaner description. 

Also, for getting a tickmark in the boxes inside the checklist, you can write it like -> [x] or [X] without any spaces. :)

**Comment by Zrahay** [comment]
Also, one more thing here. I guess your old commits are also showing up in this PR (the typo fix one). You might have made a new branch from your old branch and that could've caused this here. When you create a new branch from your old branch, the commits of that old branch carry over to the new branch since the new branch is just the copy of the old branch. 

The standard practice I believe is to go to the master branch and then create a new branch. From there, you can cherry pick the commits that you want to push to the PR. Also don't get forget to pull the latest changes while on the master branch.

I made a similar mistake yesterday so it was fresh in my mind, so just letting you know as well. :)
